node/pkg/logscribe/app.go

```diff
@@ -87,3 +163,133 @@ func bulkCopyLogs(ctx context.Context, logsChannel <-chan *[]api.LogInsertModel)
 		}
 	}
 }
+
+func ProcessLogs(ctx context.Context, service string) []LogInsertModelWithIDWithCount {
+	processedLogs := make([]LogInsertModelWithIDWithCount, 0)
+	logs, err := fetchDeleteLogs(ctx, service)
+	if err != nil {
+		log.Error().Err(err).Msg("Failed to fetch logs")
+		return nil
+	}
+	if len(logs) == 0 {
+		log.Debug().Msgf("No logs to process")
+		return nil
+	}
+
+	logHashCountMap := buildLogMap(logs)
+
+	topOccurrences := getTopOccurrences(logHashCountMap)
+	if len(topOccurrences) > topOccurrencesForGithubIssue {
+		topOccurrences = topOccurrences[:topOccurrencesForGithubIssue]
+	}
+
+	for _, pair := range topOccurrences {
+		processedLogs = append(
+			processedLogs,
+			LogInsertModelWithIDWithCount{OccurrenceCount: pair.count, LogInsertModelWithID: pair.log},
+		)
+	}
+
+	return processedLogs
+}
+
+func buildLogMap(logs []LogInsertModelWithID) map[string]LogsWithCount {
+	logHashCountMap := make(map[string]LogsWithCount) // {"hashedLog: {count, log}}
+	for _, log := range logs {
+		hash := hashLog(log)
+		logsWithCount, exists := logHashCountMap[hash]
+		if !exists {
+			logsWithCount = LogsWithCount{
+				count: 0,
+				log:   log,
+			}
+		}
+		logsWithCount.count++
+		logHashCountMap[hash] = logsWithCount
+	}
+	return logHashCountMap
+}
+
+func getTopOccurrences(hashLogPairs map[string]LogsWithCount) []LogsWithCount {
```

Comment:
let's rename it as `sortLogsWithCount` since it is not returning top occurences but returns just sorted result